### PR TITLE
8294356: IGV: scheduled graphs contain duplicated elements

### PR DESCRIPTION
**Pull request by robcasloz**
This changeset removes duplicated nodes and edges from graph dumps that include a control-flow graph:
cfg-before-after
This is achieved by ensuring that HotSpot only visits each node once when dumping IGV graphs.

#### Testing
- Tested that tens of thousands of graphs do not contain duplicated nodes or edges by instrumenting IGV and running `java -Xcomp -XX:PrintIdealGraphLevel=4`.
- Tested manually that unscheduled graphs are not affected by this changeset.
- Tested that running compiler tests with `-XX:PrintIdealGraphLevel=3` does not trigger any failure.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294356](https://bugs.openjdk.org/browse/JDK-8294356): IGV: scheduled graphs contain duplicated elements

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10423/head:pull/10423` \
`$ git checkout pull/10423`

Update a local copy of the PR: \
`$ git checkout pull/10423` \
`$ git pull https://git.openjdk.org/jdk pull/10423/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10423`

View PR using the GUI difftool: \
`$ git pr show -t 10423`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10423.diff">https://git.openjdk.org/jdk/pull/10423.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/idealGraphPrinter.cpp
Comment (chhagedorn) [inline src/hotspot/share/opto/idealGraphPrinter.cpp:763]: While at it, it might be good to replace this check with an explicit `!= nullptr` instead.

File: src/hotspot/share/opto/idealGraphPrinter.cpp
Comment (chhagedorn) [inline src/hotspot/share/opto/idealGraphPrinter.cpp:748]: While at it, this empty line could be removed.

**Review by chhagedorn — APPROVED** [review]
Looks good!

**Comment by robcasloz** [comment]
> Looks good!

Thanks for reviewing, Christian. I applied your suggestions, and replaced `NULL` with `nullptr` everywhere in the touched function for consistency.

**Review by chhagedorn — APPROVED** [review]
Changes look good!

**Comment by robcasloz** [comment]
May I get a second review for this one? The changes are small and debug-only. Thanks!

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by robcasloz** [comment]
> Looks good to me.

Thanks for reviewing, Tobias!

**Pull request closed without merging**
